### ISSUE TITLE
Implicit conversions should be sugar

### ISSUE THREAD

**Issue by olafurpg** [scalameta/scalameta#978]
```scala
  1 #:: 2 #:: 3 #:: Stream.Empty
```
produces a semanticdb like this
```
[297..300): #:: => _root_.scala.collection.immutable.Stream.ConsWrapper#`#::`(Ljava/lang/Object;)Lscala/collection/immutable/Stream;.
[303..306): #:: => _root_.scala.collection.immutable.Stream.consWrapper(Lscala/Function0;)Lscala/collection/immutable/Stream/ConsWrapper;.
[309..312): #:: => _root_.scala.collection.immutable.Stream.consWrapper(Lscala/Function0;)Lscala/collection/immutable/Stream/ConsWrapper;.
[313..319): Stream => _root_.scala.package.Stream.
[320..325): Empty => _root_.scala.collection.immutable.Stream.consWrapper(Lscala/Function0;)Lscala/collection/immutable/Stream/ConsWrapper;.
```
I expected all three `#::` to be `ConsWrapper.#::` like the first symbol and not the  `consWrapper(Function0)` sugar.

**Comment by olafurpg** [comment]
Another example
```scala
1 to 10
...
[120..122): to => _root_.scala.collection.TraversableLike#flatMap(Lscala/Function1;Lscala/collection/generic/CanBuildFrom;)Ljava/lang/Object;.
```

**Comment by olafurpg** [comment]
Fixed in #988
